CMakeLists.txt

```diff
@@ -14,18 +14,40 @@ set(CMAKE_MACOSX_RPATH 1)
 set(CMAKE_VERBOSE_MAKEFILE ON)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
 
+if(APPLE)
+  set(pdext ".pd_darwin")
+elseif(WIN32)
+  set(pdext ".dll")
+else()
+  set(pdext ".pd_linux")
+endif()
+
+set (PD_EXTENSION "${pdext}" CACHE STRING "the external extension (.pd_linux, .linux_amd64_64.so)")
+set (PD_FLOATSIZE 32 CACHE STRING "the floatsize of Pd (32 or 64)")
+
+if(PD_FLOATSIZE STREQUAL 64)
+  add_definitions(-DPD_FLOATSIZE=64)
+  # TODO check if the extension matches ".*_*_64.*"
+  if(NOT PD_EXTENSION MATCHES "^\.[^.]*_[^.]*_64\.[^.]*$")
+    message(FATAL_ERROR " double-precision externals must have an extension like '.linux_amd64_64.so'")
+  endif()
+elseif(PD_FLOATSIZE STREQUAL 32)
+else()
+  message(FATAL "FLOATSIZE must be 32 or 64")
+endif()
+
+
+
 find_path(PD_HEADER_PATH m_pd.h HINTS ${CMAKE_SOURCE_DIR}/src)
 find_library(PD_LIBRARY pd.dll)
 find_package(CSOUND)
 
 if(PD_HEADER_PATH)
-    if(APPLE) 
-        set(pdname "csound6~.pd_darwin")
-    elseif(WIN32)
-      set(pdname "csound6_tilde.dll")
+    if(WIN32)
+      set(pdname "csound6_tilde${PD_EXTENSION}")
```

Comment:
the windows external should really be called `csound6~.dll` (like with the other OSs, that is: `csound6~${PD_EXTENSION)`).

since i haven't been able to test a build on Windows (and this might actually work around some weirdo glitch) i didn't dare to touch it.
however, if possible you might think about fixing this...

Reply:
think it would be no harm to change the windows name.